db/version_edit_handler.cc

```diff
@@ -908,6 +937,7 @@ Status VersionEditHandlerPointInTime::MaybeCreateVersion(
     } else if (!s.ok()) {
       break;
     }
+    found_files.insert(file_num);
```

Comment:
nit: it's not straightforward to understand why even when s is not found, the file also added to `found_files`. I guess it's best to do this in case the file is still linked, so we can still delete the link. Would you mind adding a comment to explain that?

Reply:
That's actually a good point. The file should not be added if s is not found.